packages/spacecat-shared-gpt-client/src/clients/firefall-client.js

```diff
@@ -161,23 +167,125 @@ export default class FirefallClient {
     return jobStatusResponse;
   }
 
-  async fetch(prompt) {
+  /**
+   * Fetches data from Firefall Chat Completion API.
+   * @param prompt The text prompt to provide to Firefall
+   * @param options The options for the call:
+   *                - imageUrls: An array of URLs of the images to provide to Firefall
+   * @returns {Object} - AI response
+   */
+  async fetchChatCompletion(prompt, options = {}) {
+    const getBody = () => {
+      const { imageUrls, responseFormat, model: llmModel = 'gpt-4-turbo' } = options || {};
+      const hasImageUrls = imageUrls && imageUrls.length > 0;
```

Comment:
as `imageUrls` is accessed from the parent `options` the destructuring and validation should occur outside of the `getBody` function for efficiency reasons.